core/charm/repository/sanitizer.go

```diff
@@ -40,5 +40,12 @@ func sanitizeCharmOrigin(received, requested corecharm.Origin) (corecharm.Origin
 	}
 	result.Platform.OS = strings.ToLower(result.Platform.OS)
 
+	// Another problem area is the origin channel. We desire a full channel (track and risk)
+	// but if the effective channel implicitly resolves to a 'latest' channel, charmhub will
+	// just return the risk. This only happens for 'latest'. If we have no track, it must be latest
+	if result.Channel != nil && result.Channel.Track == "" {
```

Comment:
what happens if `result.Channel` is nil?

Reply:
Sometimes we have no channel, for instance, local charm origins. In this case, we should keep the channel nil

Reply:
Local charms should not hit this code path.

You can have a nil channel if you've requested to resolve a charm by revision. Revisions are independent from channels as the same charm revision can be in channels.